src/main/java/gift/service/WishService.java

```diff
@@ -1,78 +1,86 @@
 import gift.entity.Wish;
 import gift.exception.wish.WishNotFoundException;
-import gift.mapper.WishMapper;
 import gift.repository.WishRepository;
+import gift.util.mapper.WishMapper;
 import java.util.List;
+import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
 
 @Service
 public class WishService {
     private final WishRepository wishRepository;
     private final ProductService productService;
+    private final UserService userService;
 
-    public WishService(WishRepository wishRepository, ProductService productService) {
+    public WishService(WishRepository wishRepository, ProductService productService, UserService userService) {
         this.wishRepository = wishRepository;
         this.productService = productService;
+        this.userService = userService;
     }
 
-    public List<WishResponseDto> getWishes(Long userId) {
-        List<WishResponseDto> wishes = wishRepository.findByUserId(userId);
+    public List<WishResponse> getWishes(Long userId) {
+        List<Wish> wishes = wishRepository.findByUserId(userId);
 
-        if (wishes == null) {
-            throw new WishNotFoundException("해당 사용자의 위시 리스트가 존재하지 않습니다.");
+        if(wishes == null || wishes.isEmpty()) {
+            throw new WishNotFoundException("위시리스트가 존재하지 않습니다.");
         }
 
-        return wishRepository.findByUserId(userId);
+        return wishes.stream()
+            .map(WishMapper::toResponse)
+            .collect(Collectors.toList());
     }
 
-    public List<WishResponseDto> addWish(Long userId, WishRequestDto wishRequest) {
-        Wish wish = WishMapper.toWish(userId, wishRequest);
+    public Long addWish(Long userId, AddWishRequest request) {
+        Product product = productService.getProductById(request.productId());
+        User user = userService.getUserById(userId);
 
-        productService.checkProductExist(wish.productId());
+        Wish wish = Wish.builder()
+            .quantity(request.quantity())
+            .user(user)
+            .product(product)
+            .build();
 
-        return wishRepository.insert(wish);
+        Wish savedWish = wishRepository.save(wish);
+        return savedWish.getId();
     }
 
-    public List<WishResponseDto> updateWishes(Long userId, List<WishRequestDto> wishRequests) {
-        for (WishRequestDto wishRequest : wishRequests) {
-            Long wishId = getWishId(userId, wishRequest.productId());
-            Wish wish = WishMapper.toWish(wishId, userId, wishRequest);
+    public void updateWishes(List<UpdateWishRequest> requests) {
+        for (UpdateWishRequest request : requests) {
+            Wish wish = getWish(request.id());
+            wish.setQuantity(request.quantity());
             updateWish(wish);
         }
-        return wishRepository.findByUserId(userId);
     }
 
-    public void updateWish(Wish wish) {
-        if (wish.quantity() <= 0) {
+    private void updateWish(Wish wish) {
+        if (wish.getQuantity() <= 0) {
```

Comment:
get으로 꺼내서 값을 확인하기보다, wish에게 역할을 위임해보면 어떨까요?
```suggestion
        if (wish.isQuantityZero()) {
```
`규칙 9: getter/setter/프로퍼티를 쓰지 않는다.`

Reply:
필드 정보 검증 로직은 엔티티 내에서 만들어보겠습니다!